pydevices/HtsDevices/acq2106_423st.py

```diff
@@ -132,34 +128,31 @@ def truncate(f, n):
 
             decimator = lcma(self.decim)
 
-            if self.seg_length % decimator:
-                self.seg_length = (self.seg_length // decimator + 1) * decimator
-
-            self.dims = []
-            for i in range(self.nchans):
-                self.dims.append(Range(0., truncate((self.seg_length-1)*dt,3), dt*self.decim[i]))
-
             self.device_thread.start()
 
             segment = 0
+            begin   = 0.0
             running = self.dev.running
             max_segments = self.dev.max_segments.data()
             while running.on and segment < max_segments:
                 try:
                     buf = self.full_buffers.get(block=True, timeout=1)
-                except Queue.Empty:
+                except Empty:
                     continue
 
                 buffer = np.frombuffer(buf, dtype='int16')
                 i = 0
                 for c in self.chans:
                     if c.on:
                         b = buffer[i::self.nchans*self.decim[i]]
-                        c.makeSegment(self.dims[i].begin, self.dims[i].ending, self.dims[i], b)
-                        self.dims[i] = Range(self.dims[i].begin + truncate(self.seg_length*dt,3), self.dims[i].ending + truncate(self.seg_length*dt,3), dt*self.decim[i])
+                        
+                        dim_limits=[begin, begin + self.seg_length*dt - 1]
```

Comment:
You should compute `begin` from `segment` to avoid floating point issues when you accumulate `begin` over time